### PR TITLE
Fix counting already connected peers

### PR DESCRIPTION
**Pull request by da-kami**
In some cases we are not notified that a connection was dropped by the underlying network stack.
Thus, it can happen that we just replace a `peer_info` in the hashmap without removing it first.
Incrementing the gauge on every insert can thus lead to counting connections for one peer multiple times.
We now ensure that we only increment if the `peer_info` was not recorded already.

### CODE REVIEW

File: daemon/src/identify/dialer.rs [hunk 1]
```diff
@@ -76,13 +76,17 @@ impl Actor {
...
-        NUM_LIBP2P_CONNECTIONS_GAUGE
-            .with(&HashMap::from([
-                (WIRE_VERSION_LABEL, wire_version.as_str()),
-                (DAEMON_VERSION_LABEL, daemon_version.as_str()),
-                (ENVIRONMENT_LABEL, environment.to_string().as_str()),
-            ]))
-            .inc();
+        // Only increment if we don't know the peer info already because sometimes we are not
+        // notified about ConnectionDropped
+        if self.peer_infos.get(&peer_id).is_none() {
```
Comment (bonomat) [inline daemon/src/identify/dialer.rs]: 🙃 You could do the same thing a few lines below when inserting into the map: 
 
From rust doc: 
> Inserts a key-value pair into the map.
If the map did not have this key present, None is returned.
If the map did have this key present, the value is updated, and the old value is returned. 

e.g. 
```
if let None = self.peer_infos.insert(peer_id, peer_info.clone()) { 
... 
}
```

**Review by bonomat — APPROVED** [review]

**Review by luckysori — APPROVED** [review]
Nice!

File: daemon/src/identify/dialer.rs (hunk 1 above)
Comment (klochowicz) [reply daemon/src/identify/dialer.rs]: maybe even `if (...).is_none` {
} 

no need to bind it with `let`

**Review by klochowicz — APPROVED** [review]

File: daemon/src/identify/dialer.rs (hunk 1 above)
Comment (da-kami) [reply daemon/src/identify/dialer.rs]: I like the proposal, it's cleaner. Done!

Comment (da-kami) [reply daemon/src/identify/dialer.rs]: > 

true, could just go for `is_none` here

Comment (da-kami) [reply daemon/src/identify/dialer.rs]: clippy agrees as well 🤪

**Pull request merged**